From 1a2f82e62c4691ec0097d20292ed07749b29ec1d Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 14 Sep 2021 08:11:16 +0300
Subject: [PATCH] [SW-54874] gaudi2: arc: onix: set sram extension

On Onix, SRAM is carved out of HBM region. This patch changes
SRAM_LSB_ADDR and SRAM_MSB_ADDR values for ARC1 so that ARC1
points to correct SRAM_BASE.

Change-Id: I6e742977cbc7c5b9fd3fb964c37fadf744e4e45f
---

diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index a5d3c9f..886308a 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -236,6 +236,7 @@
 __weak void ppboot_init_stack_mon_dev_specific(void) {}
 __weak void ppboot_check_stack_mon_dev_specific(void) {}
 __weak void preboot_delay(void) {}
+__weak void sram_set_extension(void) {}
 
 void ppboot_init_stack_monitor(void)
 {
@@ -282,6 +283,8 @@
 	hl_plat_init();
 
 	preboot_delay();
+	sram_set_extension();
+
 	ret = ppboot_load_preboot_img(&loadaddr);
 	ppboot_check_stack_monitor();
 	if (ret)
diff --git a/arch/gaudi2/arc/onix.c b/arch/gaudi2/arc/onix.c
index 50a5fec..1d108dd 100644
--- a/arch/gaudi2/arc/onix.c
+++ b/arch/gaudi2/arc/onix.c
@@ -8,6 +8,7 @@
 
 #include <preboot.h>
 #include <efuse.h>
+#include <psoc_arc1_cfg_regs.h>
 
 #include "common.h"
 
@@ -40,5 +41,20 @@
 	return ret;
 }
 
+/*
+ * SRAM on Onix is mapped at address 0x1001_0002_0000_0000,
+ * change SRAM extension for ARC1 accordingly.
+ */
+#define SRAM_EXTENSION_MSB      0x10010002
+void sram_set_extension(void)
+{
+	/*
+	 * On Onix SRAM memory is carved out of HBM region,
+	 * so change the extension address for onix.
+	 */
+	regmap_writel(SRAM_EXTENSION_MSB, mmPSOC_ARC1_CFG_SRAM_MSB_ADDR);
+	regmap_writel(0x0, mmPSOC_ARC1_CFG_SRAM_LSB_ADDR);
+}
+
 void preboot_init_pcie_phy(void) {}
 void set_spi_dma_br(u16 value) {}
